"Send Crash Report User Confirmation" Dialog after restart (iOS)

alex-oreshko opened the pull request:
SetUserConfirmationHander method added to Crashes SDK.
Also Puppet shows confirmation dialog after crash and restart using this method.
Call Crashes.SetUserConfirmationHandler(UserConfirmationHandler) in Awake() method to set handler.
Crashes service starts not in AppCenterStarter.original, but later, in AppCenterBehaviour->StartAppCenter(), to give app an opportunity to set handler in Awake()

alex-oreshko committed:
"Send Crash Report User Confirmation" Dialog after restart (iOS)

SetUserConfirmationHander method added to Crashes SDK.
Also Puppet shows confirmation dialog after crash and restart using this method.
Say Crashes.SetUserConfirmationHandler(UserConfirmationHandler) in Awake() method to add handler.
Crashes service starts not in AppCenterStarter.original, but later, in AppCenterBehaviour->StartAppCenter(), to give app an opportunity to set handler in Awake()
For now array of error reports is not passed to user confirmation handler. It can be implemented later if needed.

evgeny-pol committed:
Merge branch 'develop' into SendReportsUserConfirmationDialog

alex-oreshko committed:
Send Crash Report UserConfirmation enum added

Values: DontSend(0), Send(1), AlwaysSend(2)

alex-oreshko committed:
Merge branch 'SendReportsUserConfirmationDialog' of https://github.com/alex-oreshko/AppCenter-SDK-Unity-1 into SendReportsUserConfirmationDialog

MatkovIvan commented on Assets/AppCenter/Plugins/iOS/Crashes/CrashesUnity.mm:
```diff
@@ -49,17 +50,25 @@ void appcenter_unity_crashes_disable_mach_exception_handler()
-    return [MSCrashes lastSessionCrashReport];
+    [MSCrashes setUserConfirmationHandler:^BOOL(NSArray<MSErrorReport *> *_Nonnull errorReports){
+        userConfirmationHandler();
+        return false;
```
Why do you not expose return value?
See description for this [here](https://github.com/Microsoft/AppCenter-SDK-Apple/blob/943515eac2dcbce7d2f72b0701f2d99f2df57f22/AppCenterCrashes/AppCenterCrashes/MSCrashes.h#L9)

alex-oreshko replied:
Fixed, thank you

MatkovIvan commented on Assets/Puppet/PuppetAppCenter.cs:
```diff
@@ -6,11 +6,28 @@
+        instance = this;
+    }
+
+    [MonoPInvokeCallback(typeof(Crashes.UserConfirmationHandler))]
```
Is this work on UWP?

@evgeny-pol friendly reminder about CI

alex-oreshko replied:
Not sure, a have a Mac, UWP is not supported

alex-oreshko committed:
StartCrashes method is called via Invoke to avoid linking Crash package from AppCenter

alex-oreshko committed:
Unnecessary methods removed

alex-oreshko committed:
StartCrashes() added to UWP/CrashesInternal

alex-oreshko committed:
UserConfirmationHandler returns boolean value

False - User shows confirmation UI, wait for notification
True - Discard reports

alex-oreshko committed:
User Confirmation Handler should return True if confirmation dialog should be shown

alex-oreshko committed:
ShouldProcessErrorReport added - Called by native part in case of crash after restart

Usage:  Crashes.ShouldProcessErrorReport = ShouldProcessErrorReportHandler;
Should be assigned in Awake.
Return False to discard repor, True to process report.

alex-oreshko committed:
Resolve conflicts in AppCenterBrhaviour.cs

alex-oreshko committed:
Resolving conflicts in Crashes.cs

alex-oreshko committed:
Merge branch 'develop' into SendReportsUserConfirmationDialog

alex-oreshko committed:
PuppetConfirmationDialog.meta added

alex-oreshko committed:
Merge branch 'SendReportsUserConfirmationDialog' of https://github.com/alex-oreshko/AppCenter-SDK-Unity-1 into SendReportsUserConfirmationDialog

alex-oreshko committed:
PuppetConfirmationDialog.cs.meta added

alex-oreshko committed:
spce removed

alex-oreshko committed:
Crashes.Initialize() is now called by AppCenterBehaviour.cs, not by user

alex-oreshko committed:
Crahses is initialuzed with AppCenterBehavior.InitializingServices

alex-oreshko committed:
StartAppCenter() is called in Start() instead of Awake() in AppCenterBehevior.cs

We want to be sure that user can add handlers in Awake()

alex-oreshko committed:
Resolving conflicts

Adding ConfirmationDialog again

alex-oreshko committed:
Merge branch 'develop' into SendReportsUserConfirmationDialog

alex-oreshko committed:
Made User Confirmation Dialog modal
